lib/shopify_api/context.rb

```diff
@@ -131,7 +134,7 @@ def embedded?
 
       sig { returns(T::Boolean) }
       def setup?
-        !(api_key.empty? || api_secret_key.empty? || host_name.empty?)
+        !(api_key.empty? || api_secret_key.empty? || host_name.empty? || T.must(host).empty?)
```

Comment:
I know it's not knew but my brain can't parse conditionals like that, may we can De Morgan it to:

`!api_key.empty? && !api_secret_key.empty? && !host_name.empty? && !T.must(host).empty?)`

which reduces to:

`api_key.present? && api_secret_key.present? && host_name.present? && T.must(host).present?)`

or even

`[api_key, api_secret_key, host_name, T.must(host)].all?(&:present?)`

Reply:
good call on this goofy conditional callout. I don't love it either...

`present?` is an active support helper which if I recall, we are trying to remove from this library as a dependency.